src/DynamoCore/Extensions/IExtensionStorageAccess.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dynamo.Graph;
+
+namespace Dynamo.Extensions
+{
+    public interface IExtensionStorageAccess
+    {
+        /// <summary>
+        /// Action to be invoked when the workspace is opened. The passed extensionData dictionary is a copy of the <see cref="ExtensionData"/> data dictionary.
+        /// Modifying the extensionData from this action will not modify the stored <see cref="ExtensionData"/>.
```

Comment:
is that actually true? Do you make a copy of the data first to prevent modifications to the underlying dictionary? This would be a good test.

Reply:
yes making a copy here https://github.com/DynamoDS/Dynamo/pull/11347/files#diff-d730c0ae93cb1da6c95f0ba453dd94553067feddaedf354cb63a6bfcde71ffcfR990-R991, and passing that to the `OnWorkspaceOpen`